game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/EndPoint.java

```diff
@@ -24,10 +26,9 @@
   // the next number we can run
   private long currentRunnableNumber = 0;
   private final Object numberMutex = new Object();
-  private final Object implementorsMutex = new Object();
   private final String name;
   private final Class<?> remoteClass;
-  private final List<Object> implementors = new ArrayList<>();
+  private final Set<Object> implementors = new CopyOnWriteArraySet<>();
```

Comment:
nit, I'd assign this to `Collection`. I've found that in cases where we pass `Set` to places that want a `List` parameter but could be a collection, we incur a collection copy unnecessarily.

EG:

```
foo(List.of(set));
void foo(List thisDoesNotHaveToBeList);

vs:

foo(setPassedAsCollection)
void foo(Collection anyCollectionWillDo);
```